runtime/lua/vim/shared.lua

```diff
@@ -366,7 +366,7 @@ local function can_merge(v)
 end
 
 --- Recursive worker for tbl_extend
---- @param behavior 'error'|'keep'|'force'
+--- @param behavior 'error'|'keep'|'force'|fun(key:any, v:any, prev_value:any): any
```

Comment:
```suggestion
--- @param behavior 'error'|'keep'|'force'|fun(key:any, prev_value:any, new_value:any): any
```

Reply:
swapping the order seems surprising. e.g. with the current signature, handlers that don't care about the previous value don't need to define that parameter, whereas with the swapped order, they will always have to define a (unused) `prev_value`.

Reply:
Firstly this it to be consistent with the signature of `on_bytes` in `nvim_buf_attach` which has

```lua
function on_bytes(
  bufnr,
  changed_tick,
  start_row,
  start_col,
  start_byte,
  old_row,
  old_col,
  old_byte,
  new_row,
  new_col,
  new_byte
)
```

Secondly, if you are providing custom merge semantics, then you absolutely should care about the prev value, if they don't why are they passing a function instead of just `force`? If they purely want to emulate `force` then yes they need to provide an unused param, not a big deal. Likewise, if they want to emulate `keep` then the opposite is true.

Reply:
`nvim_buf_attach` is an old interface that only exists because at the time, events (autocmds) didn't support Lua callbacks. It will be deprecated.



> Secondly, if you are providing custom merge semantics, then you absolutely should care about the prev value, if you don't why are you passing a function instead of just `force`?

True.

Reply:
> nvim_buf_attach is an old interface that only exists because at the time, events (autocmds) didn't support Lua callbacks. It will be deprecated.

With what?